ambition_utils/rrule/models.py

```diff
@@ -41,24 +52,48 @@ def handle_overdue(self, **filters):
         # Bulk update the next occurrences
         RRule.objects.update_next_occurrences(rrule_objects=rrules)
 
-    def overdue_handler_class_instances(self, **filters):
+    def process_related_model_handlers(self):
+        # Get the rrule objects that are overdue and need to be handled
+        rrule_objects = self.get_queryset().filter(
+            next_occurrence__lte=datetime.utcnow(),
+            related_object_handler_name__isnull=False,
+            related_object_id__isnull=False,
+        ).prefetch_related('related_object')
+
+        rrules_to_advance = []
+        for rrule_object in rrule_objects:
+            if hasattr(rrule_object.related_object, rrule_object.related_object_handler_name):
+                rrules_to_advance.append(
+                    getattr(rrule_object.related_object, rrule_object.related_object_handler_name)(rrule_object)
```

Comment:
Are we passing in the `rrule_object` as a param here in case the invoked method wants to do something with it before returning it?